prow/interrupts/interrupts_test.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package interrupts
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// interrupt allows for tests to trigger an interrupt as needed
+var interrupt = make(chan os.Signal, 1)
+
+// this init will be executed before that in the code package,
+// so we can inject our implementation of the interrupt channel
+func init() {
+	signalsLock.Lock()
+	signals = func() <-chan os.Signal {
```

Comment:
Couldn't you get rid of `signalsLock` by moving this into the beginning of `TestInterrupts`? Then you could also use a defer statement to undo the change (although that probably doesn't matter in this case).